transactor/src/test/scala/io/mediachain/transactor/JournalCommitSpec.scala

```diff
@@ -0,0 +1,117 @@
+package io.mediachain.transactor
+
+import org.specs2.specification.{AfterAll, BeforeAll}
+
+import java.util.function.Consumer
+import java.util.concurrent.{BlockingQueue, LinkedBlockingQueue, TimeUnit}
+
+import io.atomix.copycat.client.CopycatClient
+import io.atomix.catalyst.transport.Address
+
+import Types._
+import StateMachine._
+
+object JournalCommitSpec extends io.mediachain.BaseSpec
+  with BeforeAll
+  with AfterAll
+{
+
+  def is =
+    sequential ^
+  s2"""
+  JournalStateMachine publishes commit events:
+   - when inserting an entity $insertEntity
+   - when extending the entity chain $extendEntityChain
+   - when extending the entity chain further $extendEntityChainFurther
+   - when inserting an artefact $insertArtefact
+   - when extending the artefact chain $extendArtefactChain
+  """
+
+  def beforeAll() {
+    JournalCommitSpecContext.setup()
+  }
+  
+  def afterAll() {
+    JournalCommitSpecContext.shutdown()
+  }
+  
+  def insertEntity = {
+    val context = JournalCommitSpecContext.context
+    val res = context.dummy.client.submit(JournalInsert(Entity(Map()))).join()
+    res.foreach((entry: CanonicalEntry) => {context.ref = entry.ref})
```

Comment:
Ok, after further thought, I think the reason this whole approach feels weird to me is that it's neither unit nor integration, but kind of inbetween. I think we can express the things you want here in a better way; I would:

* set up the context (w/o the extra initializer as per #37)
* in BeforeAll, perform all your updates, receiving the updates in a queue (queue and DummyStore are your state accumulators, don't need any vars)
* in one test, drain the queue and make sure it contains all the objects you expect (you can use a matcher like `Seq(1, 2, 3, 4) must contain(allOf(2, 4))`)
* in another test, request the chain for the entity and make sure it has the right objects in the right order
* etc

I think this will make a fine integration test. Separately, we can write unit tests that test methods like `JournalStateMachine.update` in isolation, without involving clients